src/infra/src/queue/nats.rs

```diff
@@ -64,17 +64,35 @@ impl Default for NatsQueue {
     }
 }
 
+impl From<RetentionPolicy> for jetstream::stream::RetentionPolicy {
+    fn from(retention_policy: RetentionPolicy) -> Self {
+        match retention_policy {
+            RetentionPolicy::Interest => jetstream::stream::RetentionPolicy::Interest,
+            RetentionPolicy::Limits => jetstream::stream::RetentionPolicy::Limits,
+        }
+    }
+}
+
 #[async_trait]
 impl super::Queue for NatsQueue {
     async fn create(&self, topic: &str) -> Result<()> {
+        self.create_with_retention_policy(topic, RetentionPolicy::Limits)
+            .await
+    }
+
+    async fn create_with_retention_policy(
```

Comment:
Here is not enough, we changed this part, we are using create_with_max_age now. please use the latest file.